test/Microsoft.Azure.Devices.Edge.Test/SetupFixture.cs

```diff
@@ -22,7 +22,7 @@ public async Task BeforeAllAsync()
         {
             using var cts = new CancellationTokenSource(Context.Current.SetupTimeout);
             CancellationToken token = cts.Token;
-            Option<Registry> bootstrapRegistry = Option.Maybe(Context.Current.Registries.First());
+            Option<Registry> bootstrapRegistry = Option.Maybe(Context.Current.Registries.FirstOrDefault());
```

Comment:
1. Given that 1.0.9 is going to be replaced with 1.0.10, do you think it would be reasonable to update the `GetAndValidateRegistries()` in `Context.cs` ensuring that at least a register is provided? 
2. The `SasManualProvisioningFixture` and `X509ManualProvisioningFixture` also uses `Context.Current.Registries` which would likely to be running into an exception if the default registry is not provied.

Reply:
1. You don't need to provide any registry information in context.json. If you don't, the end-to-end tests will run using publicly-available mcr.microsoft.com images, which should work just fine. In fact, that's what the service deployment pipeline does; it uses publicly-released images.
2. True, those two fixtures also reference `Context.Current.Registries`, but they don't assume the IEnumerable has one or more items, so they don't run into this problem.